### ISSUE TITLE
Where should we enforce/check ForceField.aromaticity_model?

### ISSUE THREAD

**Issue by j-wags** [openforcefield/openforcefield#206]
Keeping track of my thoughts so we can come back to this: 

### Current Implementation

Both `Topology` and `ForceField` have `aromaticity_model` attributes. These are set to `OEAroModel_MDL` by default, and will raise an exception if the user (or OFFXML file) tries to set them to anything else. We've also hard-coded MDL aromaticity assignment when creating OFFMols using `from_openeye` and `from_rdkit`. The current implementation is quick and dirty, and won't actually support other aromaticity models if asked.
 
### Proposed Implementation

I think the best thing to do would be to validate molecules against the aromaticity model at the beginning of system parameterization. The specific aromaticity model would come from the value read from a SMIRNOFF data source. If multiple data sources were read, the `ForceField` object would ensure that they all specified the same aromaticity model. It seems best thar aromaticity would be checked at parameterization-time. Any sooner, and the user could change the requested aromaticity model. Any later, and we wouldn't match SMIRKS correctly. This check would first go through `Topology.check_aromaticity(aro_model)`, which would loop over all reference molecules and run `Molecule.check_aromaticity(aro_model)` on each.

`Molecule.check_aromaticity(aro_model)` would perform a round trip using the highest priority available cheminformatics toolkit (OE > RDK, presently). The toolkit would be set to raise an Exception if the molecule does not make sense given the requested aromaticity model.

### Remaining Questions
* Does it make sense for a Topology to "enforce" an aromaticity model for all of its molecules separate from/before system parameterization? It's not clear that this would save any time vs. just checking each molecule when `ForceField.create_system` is called. Checking sooner would also make it more complicated to define molecules using the API.

**Comment by jchodera** [comment]
The only time we need to use aromaticity models are when we perform operations on `Molecule` objects that require it. Since we will let molecules and topologies be edited up until that point, they may exist in a non-chemical state up until that point, so it's best not to do any sort of checking if we can avoid it---except when desired to catch errors earlier.

I'm not sure we even _want_ `Molecule.check_aromaticity()` or `Topology.check_aromaticity()` methods---and it doesn't even make sense to me what that means. Instead, when we are performing an operation that requires the molecule be perceived with a particular aromaticity model, such as finding SMARTS matches, we want to make sure the molecule has been perceived with the desired aromaticity model. 

TL;DR: The _only_ time we want to use aromaticity model information is when performing operations on a `Molecule` that require it.

**Comment by davidlmobley** [comment]
I agree with John here. And... the aromaticity model will be specified by the forcefield; it is otherwise irrelevant (and not something applied to a molecule, EXCEPT when we require it to do an operation on that molecule).

**Comment by j-wags** [comment]
OK. I'm in agreement that a `check_aromaticity` model is poorly specced. However I keep running into ambiguities in my understanding around aromaticity in general, which model we assume where, and how much of a defining feature of molecules we consider it to be. This has led me to play "whack a mole" with aromaticity perception (leading to a pretty inconsistent system of aromaticity handling implemented in the toolkit), and I've finally drafted a proposal for where I smell smoke in the current toolkit implementation and what we can do about it at #697.
